frontend/src/main.ts

```diff
@@ -3,12 +3,21 @@ import App from "./App.vue";
 import router from "./router";
 import store from "./store";
 import vuetify from "./plugins/vuetify";
+import axios from "axios";
 
 Vue.config.productionTip = false;
 
-new Vue({
+const vue = new Vue({
   router,
   store,
   vuetify,
   render: (h) => h(App),
 }).$mount("#app");
+
+axios.defaults.baseURL = `http://${process.env.VUE_APP_BACKEND_HOST}:${process.env.VUE_APP_BACKEND_PORT}`;
+
+axios.interceptors.request.use((config) => {
+  config.headers = config.headers || {};
+  config.headers["Authorization"] = "Bearer " + vue.$store.getters.token;
+  return config;
+});
```

Comment:
This seems like it will be great for the project. I hope I wasn't too negative in my other comments. I would have preferred  to see a separate issue for this, where we could have had discussion about it, but it seems like this will work really well to avoid repeatedly adding the token, specifying the host (which may be different in development and production), etc.

Also, I know that I fall into the same traps with my PRs. I don't think I'm above it. I just hope that if I annoy you about it, then you'll return the favor on my PRs.

And I just want to be crystal clear that despite all of my complaining, this is really well done. I'll be glad to be able to use it, and good work!

Reply:
No problem @tmorris42, you are right, this PR has too much scope. I wanted it to be ready for the others to have a solid basis on which to work.